Automattic/newspack-blocks
# feat(ras-acc): add name your price support to checkout button

### All Submissions:






### Changes proposed in this Pull Request:



Closes https://app.asana.com/0/1206943664367857/1206041522383357/f

This PR adds support for name your price to the checkout button block:

![Screenshot 2024-05-14 at 18 04 22](https://github.com/Automattic/newspack-blocks/assets/17905991/c8164dd4-8af5-433a-9796-e064fd3af8dd)

### How to test the changes in this Pull Request:

Note: Be sure the to checkout `epic/ras-acc` in the main newspack plugin before testing.

1. Ensure Name Your Price is installed and active
2. Create a simple product, a subscription product, and a variable product, all with name your price set, as well as any type of product with NYP NOT setup.
  - For at least one product set a maximum price 
3. Add a checkout button block for each of these products to any page
  - For at least one button set a custom price that is above the minimum price via the name your price block settings
4. Logged in as a reader, select the simple product button
5. Go through the edit billing details section (first step) of checkout and proceed to the payment section (second step)
6. Confirm you see a **Increase your support** section with a form to enter a new price for the product (pictured above)
7. Enter a price that is below the minimum price and submit. Confirm an error appears asking you to input an amount greater than the base price
8. If this is the product with a custom price set via block settings, enter a price that is below the set price listed at the top of the checkout modal, but above the minimum price and submit. Confirm an error appears asking you to input an amount greater than the base price
9. If this is the product with a maximum price set product settings, enter a price that is above the max price and submit. Confirm an error appears asking you to input a price that is less than the max price.
10. Enter a valid price and submit. Confirm the price updates at the top of the checkout modal.
11. Complete checkout and confirm the correct price was charged in WP-Admin WooCommerce Orders menu as well as in the customer receipt sent by Woo.
12. Repeat the above steps for the variable and subscription products and confirm the NYP form is present and functions as expected.
13. Repeat the above steps for the product that does not have NYP active. Confirm the NYP form does NOT appear during the checkout flow.
14. On any page, add a donation block
15. As a reader, make a donation via the donation block. Confirm the **Increase your support** field is not present during the checkout field

### Other information:
